wallet/wallet-rpc-lib/src/lib.rs

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) 2023 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+pub mod config;
+mod rpc;
+mod service;
+
+pub use service::{WalletHandle, WalletResult, WalletService};
+
+use std::time::Duration;
+
+use config::{WalletRpcConfig, WalletServiceConfig};
+use logging::log;
+
+use utils::shallow_clone::ShallowClone;
+
+const SHUTDOWN_TIMEOUT: Duration = Duration::from_secs(10);
+
+#[derive(thiserror::Error, Debug)]
+pub enum StartupError {
+    #[error(transparent)]
+    WalletService(#[from] service::InitError),
+
+    #[error("Failed to start RPC server: {0}")]
```

Comment:
start?

Reply:
Yea this error type is only used during startup, there is a different error type for reporting things that go wrong while the system is running.